tests/components/ruckus_unleashed/test_init.py

```diff
@@ -40,6 +57,26 @@ async def test_setup_entry_connection_error(hass):
     assert entry.state == ENTRY_STATE_SETUP_RETRY
 
 
+async def test_router_device_setup(hass):
+    """Test a router device is created."""
+    await init_integration(hass)
+
+    device_info = DEFAULT_AP_INFO[API_AP][API_ID]["1"]
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get_device(
+        identifiers={(CONNECTION_NETWORK_MAC, device_info[API_MAC])},
+        connections={(CONNECTION_NETWORK_MAC, device_info[API_MAC])},
+    )
+
+    assert device
+    assert device.manufacturer == MANUFACTURER
+    assert device.model == device_info[API_MODEL]
+    assert device.name == device_info[API_DEVICE_NAME]
+    assert device.sw_version == DEFAULT_SYSTEM_INFO[API_SYSTEM_OVERVIEW][API_VERSION]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)